scrapy/utils/httpobj.py

```diff
@@ -22,3 +22,73 @@ def urlparse_cached(request_or_response: Request | Response) -> ParseResult:
     if request_or_response not in _urlparse_cache:
         _urlparse_cache[request_or_response] = urlparse(request_or_response.url)
     return _urlparse_cache[request_or_response]
+
+
+# For scrapy.extensions.httpcache.WARCCacheStorage.
+# Because scrapy either doesn't expose or capture the HTTP status message,
+# so we reconstruct it when building WARC Response records for caching.
+HTTP_STATUS_MSGS = {
```

Comment:
`twisted.web.http.RESPONSES`

Reply:
Replaced, for completeness.. tho this PR will become its own project shortly.